src/core/models.py

```diff
@@ -125,12 +129,51 @@ def chroot_acl(self) -> str:
             - 'w' - write
             - 'a' - append
         """
-        return self.relation_data.get("chroot-acl", "cdrwa")
+        # TODO (zkclient): Remove this property and replace by "cdrwa" in self.extra_user_roles
+        acl = self.relation_data.get("chroot-acl")
+        if acl is not None:
+            warnings.warn(
+                "Using 'chroot-acl' in the databag is deprecated, use 'extra-user-roles' instead",
+                DeprecationWarning,
+            )
+
+        else:
+            acl = "cdrwa"
+
+        return acl
+
+    @property
+    def extra_user_roles(self) -> str:
+        """The client defined ACLs for their requested ACL.
+
+        Contains:
+            - 'c' - create
+            - 'd' - delete
+            - 'r' - read
+            - 'w' - write
+            - 'a' - append
+        """
+        return self.relation_data.get("extra-user-roles", self.chroot_acl)
```

Comment:
**suggestion** I didn't notice this before but I'm wondering whether we could put some more logic here to provide a cleaner signature, like `set[str]` rather than `str`. This way we could treat things a bit better, like "cda" being the same as "dca", given both are `set(c,d,a)`

anyhow, it is not critical and possibly not worth addressing this in this PR given it is copy and paste from the VM, and I would rather value having commits consistent between the two

Reply:
I somehow did not see this comment until I merged the VM PR. I just created a [ticket](https://warthogs.atlassian.net/browse/DPE-4446)